tools/windows-node-installer/pkg/resource/file.go

```diff
@@ -0,0 +1,99 @@
+package resource
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	set "github.com/deckarep/golang-set"
+)
+
+type InstallerInfo struct {
+	InstanceIDs      []string `json:"InstanceIDs"`
+	SecurityGroupIDs []string `json:"SecurityGroupIDs"`
+}
+
+const InstallerInfoFileName = "windows-node-installer.json"
+
+// AppendInstallerInfo appends instance id and security group to a json file and return error if file write fails.
+func AppendInstallerInfo(info *InstallerInfo, path string) error {
+	pastInfo, err := ReadInstallerInfo(path)
+	if err == nil {
+		info.InstanceIDs = append(info.InstanceIDs, pastInfo.InstanceIDs...)
+		info.SecurityGroupIDs = append(info.SecurityGroupIDs, pastInfo.SecurityGroupIDs...)
+	}
+	newInfo, err := json.Marshal(*info)
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(path, newInfo, 0644)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// ReadInstallerInfo reads instance id and security group from a json file and return error if file read fails.
+func ReadInstallerInfo(path string) (*InstallerInfo, error) {
+	pathExist, err := isPathExist(path)
+	if err != nil {
+		return nil, err
+	}
+	if !pathExist {
+		return nil, fmt.Errorf("%s does not exist", path)
+	}
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	var info InstallerInfo
+	err = json.Unmarshal(content, &info)
+	if err != nil {
+		return nil, err
+	}
+	return &info, nil
+}
+
+// RemoveInstallerInfo removes instance id and security group from a json file and return error if removal fails.
+func RemoveInstallerInfo(info *InstallerInfo, path string) error {
+	pastInfo, err := ReadInstallerInfo(path)
+	if err != nil {
+		return err
+	}
+	newInfo, err := json.Marshal(&InstallerInfo{
+		removeFromSlice(pastInfo.InstanceIDs, info.InstanceIDs),
+		removeFromSlice(pastInfo.SecurityGroupIDs, info.SecurityGroupIDs),
+	})
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(path, newInfo, 0644)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// isPathExist checks if path exist and return bool indication or checking error.
+func isPathExist(path string) (bool, error) {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return false, nil
+	} else if err != nil {
+		return false, err
+	} else {
+		return true, nil
+	}
+}
+
+// removeFromSlice removes the removals set of strings from setOfStrings and returns the resulting set.
+func removeFromSlice(setOfStrings, removals []string) []string {
```

Comment:
Just call the input `strings` as it is not a set. Moreover naming a variable based on its type is not advisable.